File: eRents.Domain/Entities/Image.cs
using System;
using System.Collections.Generic;

namespace eRents.Domain.Entities;

public partial class Image
{
	public int ImageId { get; set; }

	public int? ReviewId { get; set; }

	public int? PropertyId { get; set; }

	public string? FileName { get; set; }

	public byte[] ImageData { get; set; } = null!;

	public DateTime? DateUploaded { get; set; }

	public virtual Property? Property { get; set; }

	public virtual Review? Review { get; set; }
}

File: eRents.UI/e_rents_mobile/lib/models/image.dart
class ImageModel {
  final int imageId;
  final int? reviewId;
  final int? propertyId;
  final String? fileName;
  final String imageData;
  final DateTime? dateUploaded;

  ImageModel({
    required this.imageId,
    this.reviewId,
    this.propertyId,
    this.fileName,
    required this.imageData,
    this.dateUploaded,
  });

  factory ImageModel.fromJson(Map<String, dynamic> json) {
    return ImageModel(
      imageId: json['imageId'],
      reviewId: json['reviewId'],
      propertyId: json['propertyId'],
      fileName: json['fileName'],
      imageData: json['imageData'],
      dateUploaded: json['dateUploaded'] != null ? DateTime.parse(json['dateUploaded']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'imageId': imageId,
      'reviewId': reviewId,
      'propertyId': propertyId,
      'fileName': fileName,
      'imageData': imageData,
      'dateUploaded': dateUploaded?.toIso8601String(),
    };
  }
}
